ForceFeedback/ForceFeedback.Rules/MethodTooLongTextAdornment.cs

```diff
@@ -114,23 +102,41 @@ private void OnTextBufferChanged(object sender, TextContentChangedEventArgs e)
 
             _lastCaretBufferPosition = caretPosition + change.NewLength - 1;
 
-            if (_numberOfKeystrokes >= longMethodOccurence.LimitConfiguration.NoiseDistance)
-            {
-                if (!_view.TextBuffer.CheckEditAccess())
-                    throw new Exception("Cannot edit text buffer.");
+            if (_numberOfKeystrokes < longMethodOccurence.LimitConfiguration.NoiseDistance) return;
 
-                var textToInsert = "⌫";
+            if (!_view.TextBuffer.CheckEditAccess())
+                throw new Exception("Cannot edit text buffer.");
 
-                var edit = _view.TextBuffer.CreateEdit(EditOptions.None, null, "ForceFeedback");
-                var inserted = edit.Insert(change.NewPosition + change.NewLength, textToInsert.ToString());
+            const string textToInsert = "⌫";
 
-                if (!inserted)
-                    throw new Exception($"Cannot insert '{change.NewText}' at position {change.NewPosition} in text buffer.");
+            var edit = _view.TextBuffer.CreateEdit(EditOptions.None, null, "ForceFeedback");
+            var inserted = edit.Insert(change.NewPosition + change.NewLength, textToInsert);
 
-                edit.Apply();
+            if (!inserted)
+                throw new Exception($"Cannot insert '{change.NewText}' at position {change.NewPosition} in text buffer.");
 
-                _numberOfKeystrokes = 0;
-            }
+            edit.Apply();
+
+            _numberOfKeystrokes = 0;
+        }
+
+        private bool InteresstingChangedOccured(TextContentChangedEventArgs e)
+        {
+            if (WasChangeCausedByForceFeedback(e) || e.Changes.Count == 0)
+                return false;
+
+            var change = e.Changes[0];
+            // [RS] We trim the new text when checking for allowed characters, if the text has more than one character. This is, e.g. 
+            //      if the user inserted a linefeed and the IDE created whitespaces automatically for indention of the next line.
+            //      In this case, we want to ignore the generated leading whitespaces. 
+            //      In the case the user entered a whitespace directly, we do not want to trim it away. So we check the new text length. 
+            return e.Changes.Count > 0 &&
+                _allowedCharactersInChanges.Contains(change.NewText.Length == 1 ? change.NewText : change.NewText.Trim(' '));
+        }
+
```

Comment:
Code documentation is missing. Please describe the general intent of the method, parameter etc..